File: 백준/Silver/4779. 칸토어 집합/칸토어 집합.py
import sys

def kantore(n):
    if n == 0:
        return '-'
    str = kantore(n-1) + (' ' * (3**(n-1))) + kantore(n-1)
    return str

while True:
    try :
        N = int(sys.stdin.readline())
        print(kantore(N))
    except : # EOF 발생시
        break # 종료

File: 백준/Silver/4779. 칸토어 집합/칸토어 집합.cc
#include <bits/stdc++.h>
using namespace std;

string kantore(int n) {
    if (n == 0) {
        return "-";
    }
    string str = kantore(n - 1);
    for (int i=0; i<pow(3,n-1); i++){
        str += ' ';
    }
    str += kantore(n - 1);
    return str;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int N;
    while (!(cin >> N).eof()) {
        cout << kantore(N) << '\n';
    }

    return 0;
}
